providers/vultr/vultrProvider.go

```diff
@@ -134,47 +134,82 @@ func (api *vultrProvider) GetDomainCorrections(dc *models.DomainConfig) ([]*mode
 	models.PostProcessRecords(curRecords)
 
 	var corrections []*models.Correction
-	var create, delete, modify diff.Changeset
 	if !diff2.EnableDiff2 {
 		differ := diff.New(dc)
-		_, create, delete, modify, err = differ.IncrementalDiff(curRecords)
-	} else {
-		differ := diff.NewCompat(dc)
-		_, create, delete, modify, err = differ.IncrementalDiff(curRecords)
-	}
-	if err != nil {
-		return nil, err
-	}
+		_, create, delete, modify, err := differ.IncrementalDiff(curRecords)
 
-	for _, mod := range delete {
-		id := mod.Existing.Original.(govultr.DomainRecord).ID
-		corrections = append(corrections, &models.Correction{
-			Msg: fmt.Sprintf("%s; Vultr RecordID: %v", mod.String(), id),
-			F: func() error {
-				return api.client.DomainRecord.Delete(context.Background(), dc.Name, id)
-			},
-		})
-	}
+		if err != nil {
+			return nil, err
+		}
 
-	for _, mod := range modify {
-		r := toVultrRecord(dc, mod.Desired, mod.Existing.Original.(govultr.DomainRecord).ID)
-		corrections = append(corrections, &models.Correction{
-			Msg: fmt.Sprintf("%s; Vultr RecordID: %v", mod.String(), r.ID),
-			F: func() error {
-				return api.client.DomainRecord.Update(context.Background(), dc.Name, r.ID, &govultr.DomainRecordReq{Name: r.Name, Type: r.Type, Data: r.Data, TTL: r.TTL, Priority: &r.Priority})
-			},
-		})
-	}
+		for _, mod := range delete {
+			id := mod.Existing.Original.(govultr.DomainRecord).ID
+			corrections = append(corrections, &models.Correction{
+				Msg: fmt.Sprintf("%s; Vultr RecordID: %v", mod.String(), id),
+				F: func() error {
+					return api.client.DomainRecord.Delete(context.Background(), dc.Name, id)
+				},
+			})
+		}
 
-	for _, mod := range create {
-		r := toVultrRecord(dc, mod.Desired, "0")
-		corrections = append(corrections, &models.Correction{
-			Msg: mod.String(),
-			F: func() error {
-				_, err := api.client.DomainRecord.Create(context.Background(), dc.Name, &govultr.DomainRecordReq{Name: r.Name, Type: r.Type, Data: r.Data, TTL: r.TTL, Priority: &r.Priority})
-				return err
-			},
-		})
+		for _, mod := range modify {
+			r := toVultrRecord(dc, mod.Desired, mod.Existing.Original.(govultr.DomainRecord).ID)
+			corrections = append(corrections, &models.Correction{
+				Msg: fmt.Sprintf("%s; Vultr RecordID: %v", mod.String(), r.ID),
+				F: func() error {
+					return api.client.DomainRecord.Update(context.Background(), dc.Name, r.ID, &govultr.DomainRecordReq{Name: r.Name, Type: r.Type, Data: r.Data, TTL: r.TTL, Priority: &r.Priority})
+				},
+			})
+		}
+
+		for _, mod := range create {
+			r := toVultrRecord(dc, mod.Desired, "0")
+			corrections = append(corrections, &models.Correction{
+				Msg: mod.String(),
+				F: func() error {
+					_, err := api.client.DomainRecord.Create(context.Background(), dc.Name, &govultr.DomainRecordReq{Name: r.Name, Type: r.Type, Data: r.Data, TTL: r.TTL, Priority: &r.Priority})
+					return err
+				},
+			})
+		}
```

Comment:
Add a `return corrections, nil` to this section then remove `else {`.  That way when the references to `pkg/diff` go away, the diff won't include a reformatted "else".

Reply:
Done, can you confirm it's okay like that?